### ISSUE TITLE
dirtree should not modify the inputs

### ISSUE THREAD

**Issue by dhs-rec** [puppetlabs/pltraining-dirtree#8]
Given the following example class:
```
class dirtree_test {
  $tree = dirtree('D:/foo/bar/baz')
  file { $tree:
    ensure => directory,
  }
}
```
When applied to a Windows node, it leads to the following output:
> Notice: /Stage[main]/Dirtree_test/File[D:\foo]/ensure: created
> Notice: /Stage[main]/Dirtree_test/File[D:\foo\bar]/ensure: created
> Notice: /Stage[main]/Dirtree_test/File[D:\foo\bar\baz]/ensure: created

Which means that dirtree modifies its input. This should not happen. It's perfectly fine on Windows to use '/' as a path separator. There's no need to replace '/' with '\\\\'.

**Comment by binford2k** [comment]
I tend to agree with you, but there are some certain edge cases in which the `\` separator is required. For example scheduled tasks or DSC resources. I'm not certain those edge cases matter in this context, so I'll consider making this change.
